opengrok-indexer/src/main/java/org/opengrok/indexer/configuration/RuntimeEnvironment.java

```diff
@@ -1300,60 +1287,19 @@ public int getTabSize() {
     }
 
     /**
-     * Gets the total number of context lines per file to show: either the last
-     * value passed successfully to {@link #setContextLimit(java.lang.Short)}
-     * or {@link Configuration#getContextLimit()} as a default.
+     * Gets the total number of context lines per file to show
      * @return a value greater than zero
      */
     public short getContextLimit() {
-        return contextLimit != null ? contextLimit :
-                (short)getConfigurationValue("contextLimit");
-    }
-
-    /**
-     * Sets the total number of context lines per file to show, or resets to use
-     * {@link Configuration#getContextLimit()}.
-     * <p>
-     * N.b. the value is not mediated to {@link Configuration}.
-     * @param value a defined value or {@code null} to reset to use the
-     * {@link Configuration#getContextSurround()}
-     * @throws IllegalArgumentException if {@code value} is not positive
-     */
-    public void setContextLimit(Short value)
-            throws IllegalArgumentException {
-        if (value < 1) {
-            throw new IllegalArgumentException("value is not positive");
-        }
-        contextLimit = value;
+        return (short)getConfigurationValue("contextLimit");
     }
 
     /**
      * Gets the number of context lines to show before or after any match:
-     * either the last value passed successfully to
-     * {@link #setContextSurround(java.lang.Short)} or
-     * {@link Configuration#getContextSurround()} as a default.
      * @return a value greater than or equal to zero
      */
     public short getContextSurround() {
-        return contextSurround != null ? contextSurround :
-                (short)getConfigurationValue("contextSurround");
-    }
-
-    /**
-     * Sets the number of context lines to show before or after any match, or
-     * resets to use {@link Configuration#getContextSurround()}.
-     * <p>
-     * N.b. the value is not mediated to {@link Configuration}.
-     * @param value a defined value or {@code null} to reset to use the
-     * {@link Configuration#getContextSurround()}
-     * @throws IllegalArgumentException if {@code value} is negative
-     */
-    public void setContextSurround(Short value)
-            throws IllegalArgumentException {
-        if (value < 0) {
-            throw new IllegalArgumentException("value is negative");
```

Comment:
if you set it to negative in the read only config, will it throw an exception to signal an error now?

Reply:
also the same for the other deduplicated value

Reply:
```
  <void property="contextLimit">                                                
    <short>-100</short>                                                         
  </void>                                                                       
  <void property="contextSurround">                                             
    <short>-100</short>                                                         
  </void> 
```
fail with:
```
$ java -cp 'distribution/target/dist/*' org.opengrok.indexer.index.Indexer -R /var/opengrok/etc/readonly_configuration.xml -s /var/opengrok/src -d /var/opengrok/data/
Oct 23, 2018 1:28:55 PM org.opengrok.indexer.configuration.Configuration read
INFO: Reading configuration from /var/opengrok/etc/readonly_configuration.xml
ERROR: java.lang.IllegalArgumentException: Invalid value for "contextSurround" - "-100". Expected value greater or equal than 0
$ java -cp 'distribution/target/dist/*' org.opengrok.indexer.index.Indexer -R /var/opengrok/etc/readonly_configuration.xml -s /var/opengrok/src -d /var/opengrok/data/
Oct 23, 2018 1:29:46 PM org.opengrok.indexer.configuration.Configuration read
INFO: Reading configuration from /var/opengrok/etc/readonly_configuration.xml
ERROR: java.lang.IllegalArgumentException: Invalid value for "contextLimit" - "-100". Expected value greater than 0
```

Reply:
nice